dataframe-csv/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readCsv.kt

```diff
@@ -0,0 +1,292 @@
+@file:JvmName("ReadCsvDeephavenKt")
+
+package org.jetbrains.kotlinx.dataframe.io
+
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.api.ParserOptions
+import org.jetbrains.kotlinx.dataframe.documentation.CommonReadDelimDocs
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.ADJUST_CSV_SPECS
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.ALLOW_MISSING_COLUMNS
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.COL_TYPES
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.COMPRESSION
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.CSV_DELIMITER
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.FILE_OR_URL_READ
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.FILE_READ
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.FIXED_COLUMN_WIDTHS
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.HAS_FIXED_WIDTH_COLUMNS
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.HEADER
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.IGNORE_EMPTY_LINES
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.IGNORE_EXCESS_COLUMNS
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.IGNORE_SURROUNDING_SPACES
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.INPUT_STREAM_READ
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.PARSER_OPTIONS
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.PARSE_PARALLEL
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.PATH_READ
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.QUOTE
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.READ_LINES
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.SKIP_LINES
+import org.jetbrains.kotlinx.dataframe.documentation.DelimParams.TRIM_INSIDE_QUOTED
+import org.jetbrains.kotlinx.dataframe.impl.io.readDelimImpl
+import java.io.File
+import java.io.FileInputStream
+import java.io.InputStream
+import java.net.URL
+import java.nio.file.Path
+import kotlin.io.path.inputStream
+
+/**
+ * @include [CommonReadDelimDocs.CsvDocs]
+ * @set [CommonReadDelimDocs.DataTitleArg] File
+ * @set [CommonReadDelimDocs.DataArg] file
+ * @include [PATH_READ]
+ * @include [CSV_DELIMITER]
+ * @include [COMPRESSION]
+ * @include [CommonReadDelimDocs.CommonReadParams]
+ */
+public fun DataFrame.Companion.readCsv(
+    path: Path,
+    delimiter: Char = CSV_DELIMITER,
+    header: List<String> = HEADER,
+    hasFixedWidthColumns: Boolean = HAS_FIXED_WIDTH_COLUMNS,
```

Comment:
Are hasFixedWidthColumns and fixedColumnWidths connected? What will happen if i provide false and non-empty list?

Reply:
Then `fixedColumnWidths` will be ignored. They are only set if `hasFixedWidthColumns == true`.

Deephaven throws an error in this case. We could also do that if you like